src/ImageSharp/PixelFormats/PixelAlphaRepresentation.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+namespace SixLabors.ImageSharp.PixelFormats
+{
+    /// <summary>
+    /// Provides enumeration of the alpha value transparency behavior of a pixel format.
+    /// </summary>
+    public enum PixelAlphaRepresentation
+    {
+        /// <summary>
+        /// Indicates that the pixel format does not contain an alpha channel.
+        /// </summary>
+        None,
+
+        /// <summary>
+        /// Indicates that the transparency behavior is premultiplied.
+        /// Each color is first scaled by the alpha value. The alpha value itself is the same
+        /// in both straight and premultiplied alpha. Typically, no color channel value is
+        /// greater than the alpha channel value.
+        /// If a color channel value in a premultiplied format is greater than the alpha
+        /// channel, the standard source-over blending math results in an additive blend.
+        /// </summary>
+        Associated,
```

Comment:
Unless we know image formats where we can report this by `Identify` this would be a completely unimplemented virtual feature for now.  I know it's me who proposed the API, but now I'm concerned exposing as-is, feels a bit misleading and confusing.

How about replacing the summary with a text *"Reserved for future use"*, and moving the docs into a comment?

Reply:
I dunno... We'd then have to have a tracking issue etc. What if someone want to implement their own pixel format type and use it to represent already premultiplied pixel data?

Reply:
If a pixel had premultiplied alpha would it break any of our processors methods? if it would then either we need to go though and add checks to any that it will break even if we don't do anything else with it or we need to fix them so they function with eather type.

Reply:
It's unlikely _I think_ that it would break anything. I couldn't be 100% sure though and since no one has raised an issue suggesting that it does so far I think we're ok.

Reply:
It will only make difference when we start utilizing it in `ResizeProcessor`. There's still some non-trivial work out there: Should we watch for the value before calling `PixelOperations.To/FromVector4`? Or should we do it right in the converter method overwriting `PixelConversionModifiers.Premultiply`?

Other than this I don't really see open questions here *according to my understanding of the feature*, but I'm generally reluctant for defining public API-s for stuff that is not even prototyped.

Reply:
Might be overthinking in this case, IDK.

Reply:
I think the best place to do the check is in `PixelConversionModifiers.Premultiply`. However, there are still per-pixel calls to `Premultiply` in the convolution API.